openapi2kong/oas3_testfiles/17-request-validator-plugin-oneOf-usage.expected.json

```diff
@@ -0,0 +1,78 @@
+{
+  "_format_version": "3.0",
+  "services": [
+    {
+      "host": "backend.com",
+      "id": "730d612d-914b-5fe8-8ead-e6aa654318ef",
+      "name": "example",
+      "path": "/path",
+      "plugins": [],
+      "port": 80,
+      "protocol": "http",
+      "routes": [
+        {
+          "id": "1446ecde-7037-5f9c-8537-8217e2a12bfa",
+          "methods": [
+            "GET"
+          ],
+          "name": "example_params-test_get",
+          "paths": [
+            "~/params/test$"
+          ],
+          "plugins": [
+            {
+              "config": {
+                "body_schema": "{}",
+                "parameter_schema": [
+                  {
+                    "explode": true,
+                    "in": "query",
+                    "name": "queryid",
+                    "required": true,
+                    "schema": "{\"oneOf\":[{\"example\":10,\"type\":\"integer\"},{\"example\":2.5,\"type\":\"number\"}],\"type\":\"number\"}",
```

Comment:
has this been tested to work with the request-validation plugin? there is a `oneOf` but also a toplevel `type`. I'd expect it to work. But we better be sure.

Reply:
Even with a top level type, our request-validator plugin doesn't parse the incoming value as that particular type.
With `oneOf` in place with two types - integer and number, if we pass a query param like "test=10" or "test=1.0", it would throw an error:
<img width="636" alt="Screenshot 2024-10-25 at 12 30 43 PM" src="https://github.com/user-attachments/assets/e52ec995-7999-42df-b6ce-608611a1cff2">

I will have to add some more validations in the building logic to handle these things.

Reply:
I think top-level type is not respected by the request-validator to parse an input with that type.
Example:
 ```
- in: query
          name: queryid
          schema:
            type: string
            oneOf:
              - type: string
                example: "10"
              - type: number
                example: 2.5
          required: true
```

If top-level type is set as string, I would expect any input to be parsed as string. However, if I send "queryid=2.5", I get an error `(value should match only one schema)`.